Infrastructure/Sources/TCAExtensions/SystemEnvironment.swift

```diff
@@ -11,28 +11,25 @@ import Foundation
 @dynamicMemberLookup
 public struct SystemEnvironment<Environment> {
     var environment: Environment
-    
+
     subscript<Dependency>(
         dynamicMember keyPath: WritableKeyPath<Environment, Dependency>
     ) -> Dependency {
         get { self.environment[keyPath: keyPath] }
         set { self.environment[keyPath: keyPath] = newValue }
     }
 
-    var mainQueue: () -> AnySchedulerOf<DispatchQueue>
-    var decoder: () -> JSONDecoder
-
     private static func decoder() -> JSONDecoder {
         let decoder = JSONDecoder()
         decoder.keyDecodingStrategy = .convertFromSnakeCase
         return decoder
     }
```

Comment:
prvé použitie decoderu je veľmi časovo náročné oproti ďalším použitiam. Tu ho vždy pri každom použití vytváraš nanovo, takže to vždy bude trvať zbytočne dlho. Navrhujem to spraviť ako lazy property namiesto funkcie.

Reply:
Predelano.